go-controller/pkg/ovn/egressfirewall_test.go

```diff
@@ -1021,6 +1075,160 @@ var _ = ginkgo.Describe("OVN EgressFirewall Operations", func() {
+								{
+									DNSName: ocpnetworkapiv1alpha1.DNSName(dnsName),
+									ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+										{
+											IP: resolvedIP,
+										},
+									},
+								},
+							},
+						})
+					startOvn(dbSetup, []v1.Namespace{namespace1}, nil)
+
+					var err error
+					// Initialize the dnsNameResolver.
+					fakeOVN.controller.dnsNameResolver, err = dnsnameresolver.NewExternalEgressDNS(fakeOVN.controller.addressSetFactory,
+						fakeOVN.controller.controllerName, true, fakeOVN.watcher)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					err = fakeOVN.controller.dnsNameResolver.Run()
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+					// Create the egress firewall object.
+					_, err = fakeOVN.fakeClient.EgressFirewallClient.K8sV1().EgressFirewalls(egressFirewall.Namespace).
+						Create(context.TODO(), egressFirewall, metav1.CreateOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+					// Create the dns name resolver object.
+					_, err = fakeOVN.fakeClient.OCPNetworkClient.NetworkV1alpha1().DNSNameResolvers(dnsNameResolver.Namespace).
+						Create(context.TODO(), dnsNameResolver, metav1.CreateOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+					// check dns address set was created along with the acl and pg.
+					addrSet, _ := addressset.GetTestDbAddrSets(
+						dnsnameresolver.GetEgressFirewallDNSAddrSetDbIDs(dnsName, fakeOVN.controller.controllerName),
+						[]net.IP{net.ParseIP(resolvedIP)})
+					addrSetUUID := strings.TrimSuffix(addrSet.UUID, "-UUID")
+					dbWithACLAndPG := getEFExpectedDb(initialData, fakeOVN, namespace1.Name,
+						"(ip4.dst == $"+addrSetUUID+")", "", nbdb.ACLActionAllow)
+					addrSetDbState := append(dbWithACLAndPG, addrSet)
+					gomega.Eventually(fakeOVN.nbClient).Should(libovsdbtest.HaveData(addrSetDbState))
+
+					// delete the egress firewall object.
+					err = fakeOVN.fakeClient.EgressFirewallClient.K8sV1().EgressFirewalls(egressFirewall.Namespace).
+						Delete(context.TODO(), egressFirewall.Name, metav1.DeleteOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+					// delete the dns name resolver object.
+					err = fakeOVN.fakeClient.OCPNetworkClient.NetworkV1alpha1().DNSNameResolvers(dnsNameResolver.Namespace).
+						Delete(context.TODO(), dnsNameResolver.Name, metav1.DeleteOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+					// check dns address set is cleaned up on delete.
+					expectedDatabaseState := getEFExpectedDbAfterDelete(dbWithACLAndPG)
+					gomega.Eventually(fakeOVN.nbClient).Should(libovsdbtest.HaveData(expectedDatabaseState))
+					return nil
+				}
+				err := app.Run([]string{app.Name})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			})
+		})
+		ginkgo.It(fmt.Sprintf("correctly creates egress firewall using dns name resolver with IPv6 address, gateway mode %s", gwMode), func() {
```

Comment:
ipv4 and ipv6 can be one test with different inputs, maybe we can use DescribeTable for all combinations of gateway modes, ip families, and dnsResolvers